pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -345,31 +255,47 @@ func (r *repository) Put(manifest *schema1.SignedManifest) error {
 				},
 			},
 			DockerImageReference: fmt.Sprintf("%s/%s/%s@%s", r.registryAddr, r.namespace, r.name, dgst.String()),
-			DockerImageManifest:  string(manifest.Raw),
+			DockerImageManifest:  string(payload),
 		},
 	}
 
-	if err := r.fillImageWithMetadata(manifest, &ism.Image); err != nil {
-		return err
+	for _, option := range options {
+		if opt, ok := option.(distribution.WithTagOption); ok {
```

Comment:
> What about defaulting to latest ?

Tag comes from the client. The client specifies `latest` as the default value. If tag is not in request, then we should generate an error.